### ISSUE TITLE
exception sending message to a dead process

### ISSUE THREAD

**Issue by flaviogrossi** [derekkraan/horde#96]
Sometimes the following exception can be raised:

```
[error] GenServer :"horde_82228185.Crdt" terminating                
** (ArgumentError) argument error                                                
    :erlang.send(:horde_82228185, {:crdt_update, [{:add, {:process, :proc_12}, {nil, %{id: :proc_12, modules: [Task], restart: :permanent, shutdown: 10, start: {Task, :start_link, [#Function<0.66858309/0 in SupervisorTest.__ex_unit_setup_0/1>]}, type: :worker}}}]})                                                           
    (horde) lib/horde/supervisor_supervisor.ex:14: anonymous fn/2 in Horde.SupervisorSupervisor.init/1                                                            
    (delta_crdt) lib/delta_crdt/causal_crdt.ex:239: DeltaCrdt.CausalCrdt.update_state_with_delta/3                                                                
    (delta_crdt) lib/delta_crdt/causal_crdt.ex:204: DeltaCrdt.CausalCrdt.handle_cast/2                                                                            
    (stdlib) gen_server.erl:637: :gen_server.try_dispatch/4                      
    (stdlib) gen_server.erl:711: :gen_server.handle_msg/6                        
    (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3                       
Last message: {:"$gen_cast", {:operation, {:add, [process: :proc_12, nil: %{id: :proc_12, modules: [Task], restart: :permanent, shutdown: 10, start: {Task, :start_link, [#Function<0.66858309/0 in SupervisorTest.__ex_unit_setup_0/1>]}, type: :worker}]}}}
```

This often happen in tests (e.g. the supervisor test of this project) and it seems to be caused by [this line](https://github.com/derekkraan/horde/blob/a0091fd45785314be62392e78ad27ea888bfe3e9/lib/horde/supervisor_supervisor.ex#L12). When it happens the process registered with the name `root_name` is already dead, and an argument error is raise by `send/2`.

A simple
```elixir
on_diffs: fn diffs ->
  try do
    send(root_name, {:crdt_update, diffs})
  rescue
    e ->
      nil
  end
```
is enough, but i dont know if this is the right approach here or there is a better, more structured solution.

What do you think?

**Comment by derekkraan** [comment]
This approach is problematic, because if a diff fails to send for whatever reason, then the supervisor / registry will go out of sync.

I think the better approach would be to allow signalling an error as a return value to the `on_diffs` callback that simply stops the delta_crdt process.

**Comment by derekkraan** [comment]
This was fixed in #126
